runtime/controller.go

```diff
@@ -673,22 +726,23 @@ func (c *Controller) safeRename(from, to *runtimev1.ResourceName) error {
 		return c.catalog.rename(from, to)
 	}
 
-	// Collision, do a delete+create instead of a rename
+	// Collision, do a create+delete instead of a rename
```

Comment:
Whats the idea behind doing it in this order?

Reply:
```suggestion
	// Collision, do a create+delete instead of a rename
	// (since creation might fail if the name is taken, whereas the delete is almost certain to succeed)
```

Reply:
Added a comment to explain – creation may uncover name conflicts, whereas deletion is basically sure to succeed